### ISSUE TITLE
[issue]: Not Booting ISOs on Macbook

### ISSUE THREAD

**Issue by 0xlne** [ventoy/Ventoy#1056]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0.49

### What about latest release

Yes. I have tried the latest release, but the bug still exist.

### BIOS Mode

UEFI Mode

### Partition Style

MBR

### Disk Capacity

16Gb, 128Gb

### Image file checksum (if applicable)

Yes.

### What happened?

**Device** - Macbook Air A1466 (Core i5).
**Context** - Use Ventoy to chainload other ISOs. Tried Windows 10 20H2, Hirens Boot CD (WinPE), Ubuntu 16.04, Linux Mint 20.04, voidlinux, Alpine Linux 3.14.1.
n.b. - tried Ventoy with both MBR & GPT partition types.
**Behaviour** - Ventoy boots, lists ISOs, but further chainloading them (any of the above) results in a black screen with a non-blinking cursor.
The same Ventoy USB boots fine on a typical laptop (Lenovo Thinkpad T420).
Other bootable USBs (using the same stick) prepared by other apps (Rufus, Balena Etcher) boot fine on the Macbook Air.

**Comment by ventoy** [comment]
Ventoy actually doesn't support Macbook. I never test it on Macbook.
You can do some try though:
1. For Ubuntu/LinuxMint..  
Press `Ctrl+r` to enter grub2 mode in Ventoy's menu, and then select the ISO to boot.

2. For standard Windows ISO
Press `Ctrl+w` to enter wimboot mode in Ventoy's menu, and then select the ISO to boot.

**Comment by 0xlne** [comment]
I'm assuming you mean the ExMenu by pressing F6?
If that's the case, it doesn't work; I get a "ventoy_grub.cfg NOT exist" message.
![IMG_20210812_232312](https://user-images.githubusercontent.com/26036905/129271998-cc360be1-db1d-4a34-8963-90ae694c4a0b.jpg)

**Comment by ventoy** [comment]
Not press F6, but `Ctrl+w` or `Ctrl+r`
This is a description about `Ctrl+w`
https://www.ventoy.net/en/doc_wimboot.html

**Comment by 0xlne** [comment]
I tried that prior to posting my pervious comment:  
- the red Wimboot mode is not displayed on the Macbook
- it is indeed displayed on a Thinkpad T420  
I am of course using the same memory stick; just moving them between PCs.

**Comment by nguyenhathaitrong** [comment]
https://mac247.vn/huong-dan-tu-cai-windows-cho-mac-de-dang-nhat

**Comment by jr1221** [comment]
Sounds like #282

**Comment by nguyenhathaitrong** [comment]
Currently ventoy does not support mac

**Comment by spiralofhope** [comment]
I respect that the author does not support a Mac.

Because others will search for and find this issue, I'll add my tests/success:

Using Ventoy `1.0.82`

With an Apple Macbook Pro A1278 - 13" Laptop (Mid 2012)

For OpenCore this Macbook is `MacBookPro9,2` and for board-id is `Mac-6F01561E16C75D06`

- Power off
- Insert USB stick with working Ventoy and ISOs.  My Macbook has two USB ports; either will work.
- Power on
- Hold `control` while powering up
- (the boot selector is seen)
- Select your USB stick.  Perhaps it is called "EFI Boot" or "Ventoy". The choice will be orange with a large USB symbol.
- Press `enter`
- When Ventoy is seen, press `control-r` (to enter GRUB2 mode, you will see that text in red at the bottom)
- Select the ISO
- Press `enter`
- Use any default boot options that distribution presents.  (Perhaps press `enter` a few times)

Tested and working with these live distributions:

- `devuan_chimaera_4.0.0_amd64_desktop-live.iso`
- `kali-linux-2021.2-live-amd64.iso`
- `linuxmint-20.1-cinnamon-64bit.iso`
- `lubuntu-20.04.2-desktop-amd64.iso` (with persistence)
- `ubuntu-20.04.2.0-desktop-amd64.iso`

This did not work with:

- `pclinuxos64-kde5-magnum-2021.0215.iso`
- `MX-21.1_ahs_x64.iso` (MX Linux)
- `MX-21.1_fluxbox_x64.iso` (MX Linux)
- `tails-amd64-4.19.iso`
- `ventoy-1.0.82-livecd.iso`
- Windows 10 (downloaded from Microsoft on 2021-06-05)
- Windows 11 - Win11_English_x64v1.iso

Note that if you are testing multiple distributions, then it is best to power off the system.  Simply rebooting may not present the USB stick as a boot option.

Using Wimboot mode (instead of `control-r`, press `control-w` from the Ventoy ISO select screen) worked for Windows, although I did not attempt to install them:

- Windows 10 (downloaded from Microsoft on 2021-06-05)
- Windows 11 - Win11_English_x64v1.iso

macOS (`Install_macOS_10.15.7-19H15.iso`) failed with both `control-r` and `control-w` although I'm uncertain if I built it right or not.

**Comment by tejasraman** [comment]
FYI, booting Linux with grub2 mode works, but booting with normal mode fails (shows a white `-` in the corner). This only works with 1.0.80 and above from my testing though.

MX did work for me with v1.0.87 and 1.0.88. Windows 10 (21h2) and Windows 11 (22H2) both booted and installed with WIMboot mode.

I tested this on a Late 2011 13" MBP.

Also @spiralofhope Ventoy doesn't support macOS ISOs at all... they are not even supposed to be ISOs as macOS stores the EFI files on a different location then the standard locations (see [Eclectic Light's](https://eclecticlight.co/2018/08/10/booting-the-mac-loading-boot-efi-and-secure-boot/) article on this.

**Comment by jaknz** [comment]
> Ventoy actually doesn't support Macbook. I never test it on Macbook. You can do some try though:
> 
> 1. For Ubuntu/LinuxMint..
>    Press `Ctrl+r` to enter grub2 mode in Ventoy's menu, and then select the ISO to boot.
> 2. For standard Windows ISO
>    Press `Ctrl+w` to enter wimboot mode in Ventoy's menu, and then select the ISO to boot.

This solved my issue of not being able to Vintoy-boot Linux ISOs on a 2013 Mac Pro. Thanks!

**Comment by tejasraman** [comment]
> > Ventoy actually doesn't support Macbook. I never test it on Macbook. You can do some try though:
> > 
> > 1. For Ubuntu/LinuxMint..
> >    Press `Ctrl+r` to enter grub2 mode in Ventoy's menu, and then select the ISO to boot.
> > 2. For standard Windows ISO
> >    Press `Ctrl+w` to enter wimboot mode in Ventoy's menu, and then select the ISO to boot.
> 
> This solved my issue of not being able to Vintoy-boot Linux ISOs on a 2013 Mac Pro. Thanks!

In versions 1.0.80 and newer those options show up without keyboard shortcuts.

**Comment by spiralofhope** [comment]
> @ventoy it seems that using grub2 makes it work, maybe it should default to it when detecting a macbook?

But the Macbook is not supported. The author does not have one or write software for it.

Perhaps someone will provide a pull request with necessary programming, but that code still might not be included since, again, Ventoy unfortunately does not support a Macbook.

Maybe someone will buy the author a Macbook?

**Comment by Martinligabue** [comment]
> > @ventoy it seems that using grub2 makes it work, maybe it should default to it when detecting a macbook?
> 
> But the Macbook is not supported. The author does not have one or write software for it.
> 
> Perhaps someone will provide a pull request with necessary programming, but that code still might not be included since, again, Ventoy unfortunately does not support a Macbook.
> 
> Maybe someone will buy the author a Macbook?

why does the author have to have access to a physical macbook to accept a community-tested proposal? Do you think Linus Torvalds has access to every existing pc and tests every change on them in order to push changes into the kernel?
I don't see why you wouldn't accept a pull request just because you don't officially support the Macbook.
It's not like the change would affect any other user in any other scenario.

**Comment by spiralofhope** [comment]
> why does the author have to have access to a physical macbook to accept a community-tested proposal?

The kernel project is not just a bunch of people, it is a network of curated strata of developers who check the code of one another.  Submissions make their way up the chain, each being examined _and tested_ before _maybe_ making it into the mainstream codebase.

Ventoy is not that.  Adopting someone else's alien and untestable code is not good practice for this project.

I have been subject to the following statement, and I hate it:  You are free to fork it.  That is what some other developer would have to do to implement Macbook features.  Whatever developer writes Macbook code for Ventoy would themselves need to have a Macbook.

Although no developer should rely on some random contributor's code and testing, it's hypothetically possible someone else's code could be so simple that the Ventoy developer could understand and trust it with no testing, so that it would not need to be forked.  That is a hypothetical though, and so no promises get made.  It's difficult to trust a stranger.

----

Some more on that topic...

Users should themselves also be wary of such code being introduced.  What if the original side-developer becomes unavailable and that code becomes orphaned?  Perhaps the user grows fond or reliant of the code.  Perhaps that code causes weird bugs later and cannot be fixed easily (since the lead developer cannot test properly).  Perhaps it becomes tangled into the codebase and can't even be removed easily.

Even projects which have multiple trusted developers end up with problems when one of the developers become unavailable.

While this has been an interesting thought experiment, I hope you understand why I will end this topic with:

"I hope this minor Macbook support is simple to be included and supported by a different developer, and could be disabled by the main developer if it causes problems in the future."

And so I hope it is understandable when _this_ developer says, politely, no.

**Comment by Martinligabue** [comment]
@spiralofhope that's now perfectly clear

**Comment by spiralofhope** [comment]
> @spiralofhope that's now perfectly clear

I hope I came across as respectful.  :)

**Comment by ibay770** [comment]
Yes you did. Thank you. And thanks for the detailed explanation. 
It boots win10 from my A1367, macbook 2011.
